Sources/performance-worker/BenchmarkService.swift

```diff
@@ -0,0 +1,161 @@
+            $0.payload = Grpc_Testing_Payload.with {
+              $0.body = Data(count: Int(message.responseSize))
+            }
+          }
+        )
+      }
+      return [:]
+    }
+  }
+
+  /// Single-sided unbounded streaming from client to server.
+  /// The server returns a payload with the size requested by the client once the client does WritesDone.
+  func streamingFromClient(
+    request: ServerRequest.Stream<Grpc_Testing_BenchmarkService.Method.StreamingFromClient.Input>
+  ) async throws
+    -> ServerResponse.Single<Grpc_Testing_BenchmarkService.Method.StreamingFromClient.Output>
+  {
+    var responseSize = 0
+    for try await message in request.messages {
+      if message.responseStatus.isInitialized {
+        try self.checkOkStatus(message.responseStatus)
+      }
+      responseSize = Int(message.responseSize)
+    }
+
+    return ServerResponse.Single(
+      message: Grpc_Testing_BenchmarkService.Method.StreamingFromClient.Output.with {
+        $0.payload = Grpc_Testing_Payload.with {
+          $0.body = Data(count: responseSize)
+        }
+      }
+    )
+  }
+
+  /// Single-sided unbounded streaming from server to client.
+  /// The server repeatedly returns a payload with the size requested by the client.
+  func streamingFromServer(
+    request: ServerRequest.Single<Grpc_Testing_BenchmarkService.Method.StreamingFromServer.Input>
+  ) async throws
+    -> ServerResponse.Stream<Grpc_Testing_BenchmarkService.Method.StreamingFromServer.Output>
+  {
+    return ServerResponse.Stream { writer in
+      if request.message.responseStatus.isInitialized {
+        try self.checkOkStatus(request.message.responseStatus)
+      }
+
+      while working.load(ordering: .acquiring) {
+        try await writer.write(
+          Grpc_Testing_BenchmarkService.Method.StreamingCall.Output.with {
+            $0.payload = Grpc_Testing_Payload.with {
+              $0.body = Data(count: Int(request.message.responseSize))
+            }
+          }
+        )
+      }
+      return [:]
+    }
+  }
+
+  /// Two-sided unbounded streaming between server to client.
+  /// Both sides send the content of their own choice to the other.
+  func streamingBothWays(
+    request: GRPCCore.ServerRequest.Stream<
+      Grpc_Testing_BenchmarkService.Method.StreamingBothWays.Input
+    >
+  ) async throws
+    -> GRPCCore.ServerResponse.Stream<Grpc_Testing_BenchmarkService.Method.StreamingBothWays.Output>
+  {
+    for try await message in request.messages {
+      if message.responseStatus.isInitialized {
+        try self.checkOkStatus(message.responseStatus)
+      }
+    }
+    return ServerResponse.Stream { writer in
+      while working.load(ordering: .acquiring) {
+        try await writer.write(
+          Grpc_Testing_BenchmarkService.Method.StreamingCall.Output.with {
+            $0.payload = Grpc_Testing_Payload.with {
+              $0.body = Data(count: 100)
```

Comment:
Sorry, was there a specific reason we used exactly 100? (We are using a single value so we can create the message once - as you pointed before I will create it before the loop, but was there a reason for this size specifically?)

Reply:
https://github.com/grpc/grpc-java/blob/36e9f0dfacc853b5d54f64bb14305f0b4c323589/benchmarks/src/main/java/io/grpc/benchmarks/qps/AsyncServer.java#L192-L199

Reply:
so because the java implementation does this? my understanding of the comment is that they are using the same size for all responses because the spec allows this, but not that the spec says it should be 100. Am I missing something?

Reply:
You're not missing anything. We're doing it because Java (and other impls) are using 100, i.e. there is precedent elsewhere and we haven't just picked 100 out of thin air (someone else did, we're just following).